$MirOS: ports/lang/lua/patches/patch-src_Makefile,v 1.2 2009/08/29 12:41:13 tg Exp $
$OpenBSD: patch-src_Makefile,v 1.5 2008/04/19 17:59:32 okan Exp $
--- src/Makefile.orig	2008-01-19 19:37:58.000000000 +0000
+++ src/Makefile	2009-08-30 21:42:29.000000000 +0000
@@ -7,12 +7,18 @@
 # Your platform. See PLATS for possible values.
 PLAT= none
 
-CC= gcc
-CFLAGS= -O2 -Wall $(MYCFLAGS)
+CFLAGS+= $(MYCFLAGS)
 AR= ar rcu
 RANLIB= ranlib
 RM= rm -f
-LIBS= -lm $(MYLIBS)
+LDLUA_SOBJS=	$(CORE_O) $(LIB_O)
+LDLUA_LDADD=	-lm
+DPLUA_A=	${LUA_A}
+LDLUA_A=	${LUA_A} ${LDLUA_LDADD}
+# RELINK=.PHONY on OSX
+DPLUA_LINK=	${LUA_SO} ${RELINK}
+LDLUA_LINK=	-L${.CURDIR} -llua
+LIBS=		${MYLIBS}
 
 MYCFLAGS=
 MYLDFLAGS=
@@ -34,9 +40,10 @@ LUA_O=	lua.o
 
 LUAC_T=	luac
 LUAC_O=	luac.o print.o
+LUA_SO=	.stamp.LUA_SO
 
 ALL_O= $(CORE_O) $(LIB_O) $(LUA_O) $(LUAC_O)
-ALL_T= $(LUA_A) $(LUA_T) $(LUAC_T)
+ALL_T= $(LUA_A) $(LUA_T) $(LUAC_T) $(LUA_SO)
 ALL_A= $(LUA_A)
 
 default: $(PLAT)
@@ -51,11 +58,23 @@ $(LUA_A): $(CORE_O) $(LIB_O)
 	$(AR) $@ $?
 	$(RANLIB) $@
 
-$(LUA_T): $(LUA_O) $(LUA_A)
-	$(CC) -o $@ $(MYLDFLAGS) $(LUA_O) $(LUA_A) $(LIBS)
+$(LUA_T): $(LUA_O) ${DPLUA_LINK}
+	$(CC) -o $@ $(MYLDFLAGS) $(LUA_O) ${LDLUA_LINK} $(LIBS)
 
-$(LUAC_T): $(LUAC_O) $(LUA_A)
-	$(CC) -o $@ $(MYLDFLAGS) $(LUAC_O) $(LUA_A) $(LIBS)
+$(LUAC_T): $(LUAC_O) ${DPLUA_A}
+	$(CC) -o $@ $(MYLDFLAGS) $(LUAC_O) ${LDLUA_A} $(LIBS)
+
+$(LUA_SO): ${LDLUA_SOBJS}
+	LDADD=${LDLUA_LDADD:Q} ${MAKE} -f ${.SYSMK:Q}/bsd.lib.mk \
+	    LIB=lua SHLIB_VERSION=${LIBlua_VERSION} _LIBS_STATIC=no \
+	    SOBJS=${LDLUA_SOBJS:Q}
+	:>$@
+
+soinstall:
+	LDADD=${LDLUA_LDADD:Q} ${MAKE} -f ${.SYSMK:Q}/bsd.lib.mk \
+	    install LIBDIR=${INSTALL_LIB:Q} \
+	    LIB=lua SHLIB_VERSION=${LIBlua_VERSION} _LIBS_STATIC=no \
+	    SOBJS=${LDLUA_SOBJS:Q}
 
 clean:
 	$(RM) $(ALL_T) $(ALL_O)
@@ -87,7 +106,7 @@ ansi:
 	$(MAKE) all MYCFLAGS=-DLUA_ANSI
 
 bsd:
-	$(MAKE) all MYCFLAGS="-DLUA_USE_POSIX -DLUA_USE_DLOPEN" MYLIBS="-Wl,-E"
+	$(MAKE) all MYCFLAGS=-DLUA_USE_BSD MYLIBS="-Wl,-E -ledit -lcurses"
 
 freebsd:
 	$(MAKE) all MYCFLAGS="-DLUA_USE_LINUX" MYLIBS="-Wl,-E -lreadline"
@@ -115,6 +134,12 @@ posix:
 solaris:
 	$(MAKE) all MYCFLAGS="-DLUA_USE_POSIX -DLUA_USE_DLOPEN" MYLIBS="-ldl"
 
+relink-aix relink-ansi relink-bsd relink-freebsd relink-generic relink-linux:
+relink-mingw relink-posix relink-solaris:
+
+relink-macosx:
+	$(MAKE) $(LUA_T) MYCFLAGS=-DLUA_USE_LINUX MYLIBS="-lreadline" RELINK=.PHONY
+
 # list targets that do not create files (but not all makes understand .PHONY)
 .PHONY: all $(PLATS) default o a clean depend echo none
 
